lib/kafka_ex/socket.ex

```diff
@@ -79,11 +79,19 @@ defmodule KafkaEx.Socket do
   end
 
   @doc """
+  Returns true if the socket is open
+  """
+  @spec open?(KafkaEx.Socket.t) :: boolean
+  def open?(socket = %KafkaEx.Socket{}) do
+    info(socket) != nil
```

Comment:
Good catch.  `Port.info` returns `nil` if the port is closed, so I just had the spec for `Socket.info/2` incorrect.